go.mod

```diff
@@ -1,27 +1,29 @@
 module github.com/btcsuite/btcwallet
 
 require (
-	github.com/btcsuite/btcd v0.22.0-beta.0.20210803133449-f5a1fb9965e4
+	github.com/btcsuite/btcd v0.22.0-beta.0.20220111222628-c5e6d3cf5f56
+	github.com/btcsuite/btcd/btcutil v1.0.1-0.20220111222628-c5e6d3cf5f56
+	github.com/btcsuite/btcd/btcutil/psbt v1.0.1-0.20220111222628-c5e6d3cf5f56
 	github.com/btcsuite/btclog v0.0.0-20170628155309-84c8d2346e9f
-	github.com/btcsuite/btcutil v1.0.3-0.20210929233259-9cdf59f60c51
-	github.com/btcsuite/btcutil/psbt v1.0.3-0.20210929233259-9cdf59f60c51
 	github.com/btcsuite/btcwallet/wallet/txauthor v1.1.0
 	github.com/btcsuite/btcwallet/wallet/txrules v1.1.0
 	github.com/btcsuite/btcwallet/wallet/txsizes v1.1.0
 	github.com/btcsuite/btcwallet/walletdb v1.3.5
 	github.com/btcsuite/btcwallet/wtxmgr v1.3.0
```

Comment:
For a Go consumer of the btcwallet module to use btcwallet as an import where the replaces here aren't applied, this version of wtxmgr is too old.  I had to update it to a pseudo version like `v1.3.1-0.20211201210108-79de92f527dc` in https://github.com/btcsuite/btcwallet/pull/782, which your PR would hopefully replace.  Ideally the maintainers would tag this a wtxmgr 1.4.0 (or whatever), but otherwise a pseudo-version is needed either here in the go.mod of the consumer.

But, the btcwallet **submodules that also require btcutil** and btcd are:
- wallet/txauthor
- wtxmgr
- wallet/txrules

I think I would update these submodules requires first, then come back to this main btcwallet go.mod to have it point to new pseudo-versions of these updated submodules.

Reply:
Feedback appreciated; I'm currently a bit stuck ironing out all these in a way that makes everything align and builds when imported in lnd. Hopefully I'll make some progress.

Reply:
I *think* I've addressed all your points now. All checks seem to work out as well. Unfortunately I have not been able to figure out a way to get everything to line up without also simultaneously using an updated version of neutrino (and same thing there, making this a bit cyclical).

Would be really good if we can make some further progress here so any assistance to get this mergable is very much appreciated.

Reply:
Just to clarify that I'm not a maintainer here, just another fellow consumer with an interest in seeing the updates.

In general I think the way forward needs to involve a little back and forth between the repos.  With the wallet submodules (wallet/txauthor, wtxmgr, wallet/txrules) updated for the new btcutil, that would let neutrino update it's requires on those wallet submodules (in https://github.com/lightninglabs/neutrino/pull/242, but without replaces), then come back to btcwallet and update it's neutrino require.

These gymnastics are happening because btcd is still v0.x, not yet following semantic versioning.  **The btcutil move is a breaking change to consumers with btcutil types in their exported APIs**, and normally in Go land that would require a change to the semantic import versioning (SIV) in such consumers, e.g. btcd/v2 so that btcd consumers are not borken.

Unfortunately, the same breaking change I've described in btcd is about to happen to these three wallet submodules that are v1.x.  So, it all depends on how the maintainers want to handle this (allow breaking changes thus violating SIV or bump these wallet submodules to /v2 and update any requires on them).

Reply:
> Just to clarify that I'm not a maintainer here, just another fellow consumer with an interest in seeing the updates.

That was clear - the request for assistance was more a shout in the void than directly addressing you personally, though of course good to see you're still interested in this as well :)

I think I finally got what you were already trying to say above. Taking a new shot for the first step of that lock-step change. Really hoping maintainers will pop in here soon as quite a lot of things are blocked by this right now.